### PR TITLE
test: Handle React form inputs

### PR DESCRIPTION
**Pull request by mvollmer**

### CODE REVIEW

File: test/common/testlib.py [hunk 1]
```diff
@@ -231,6 +231,11 @@ def key_press(self, keys):
             else:
                 self.cdp.invoke("Input.dispatchKeyEvent", type="char", text=k)
 
+    def set_input_text(self, selector, val):
```
Comment (mareklibra) [inline test/common/testlib.py]: I would explicitly mention "type" in the name of this function

File: pkg/storaged/dialogx.jsx [hunk 2]
```diff
@@ -280,7 +280,7 @@ export const PassInput = (tag, title, options) => {
         initial_value: "",
 
         render: (val, change) =>
-            <input data-field={tag}
+            <input data-field={tag} data-field-type="text-input"
```
Comment (mareklibra) [inline pkg/storaged/dialogx.jsx]: As mentioned in other PRs, the `data-*` attributes are not preferred in React. They exist just for compatibility with (older) libraries.
Wouldn't it be better to determine the type form css `class`? If not in all cases recently, we can add a new one, without it's definition.

Comment (mvollmer) [reply pkg/storaged/dialogx.jsx]: We might use a class for `data-field-type`, but it's more difficult for `data-tag`.  But I don't see why we should stop using `data-*` attributes.  They work, React doesn't warn about them, and there isn't a equivalent alternative, right?

File: test/common/testlib.py (hunk 1 above)
Comment (mvollmer) [reply test/common/testlib.py]: I'd say "typing" is an implementation detail.  If we later find a better way that doesn't involve key-press events, I think we should use it.

**Comment by martinpitt** [comment]
Let's merge this now, as there are quite a lot of other React PRs and these tend to age quickly. It doesn't make the code any worse, and we can iterate on it again.

**Pull request closed without merging**